lib/train/transports/local.rb

```diff
@@ -230,6 +231,12 @@ def acquire_pipe
 
           pipe = nil
 
+          # 4. Verify ownership before connecting
+          owner, current_user, is_owner = pipe_owned_by_current_user?(pipe_name)
+          unless is_owner
+            raise PipeError, "Unauthorized user '#{current_user}' tried to connect to pipe '#{pipe_name}'. Pipe is owned by '#{owner}'."
+          end
```

Comment:
but we just created with the right privileges .. why do we have to check this?

Reply:
As recommended in the shared remediation, even if we create the pipe with the correct privileges and ACLs, verifying ownership before connecting is a critical defense-in-depth measure. This helps prevent race condition and pre-creation attacks, ensuring that our process only connects to pipes it actually owns.